### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by nowhszh**
1. 放弃使用全局变量
2. 封装复用日志处理代码
3. 高亮关键日志
4. 添加错误返回值检查，异常终止

### CODE REVIEW

File: pyextract.py [hunk 1]
```diff
@@ -102,139 +97,183 @@ def __set__(self, instance, value):
...
-        dirs = os.listdir(path)
-        for file in dirs:
-            if '.gz' in file:
-                filename = file.replace(".gz", "")
-                gzip_file = gzip.GzipFile(path + '/' + file)
-                with open(path + "/" + filename, 'wb+') as f:
-                    f.write(gzip_file.read())
-    print("gunzip all finish")
 
+def CHECK_EXIST(ret):
```
Comment (nowhszh) [inline pyextract.py]: error function name

**Review by Junbo-Zheng — CHANGES_REQUESTED** [review]
@nowhszh Please update README.md too.

File: pyextract.py [hunk 2]
```diff
@@ -9,21 +9,18 @@
...
-        self.output_file = "file.tar.gz"
-        self.merge_file = "./merged.log"
-        self.filter_pattern = "log\\d*|tmp.log"
-        self.tmp_log = "tmp.log"
-
-
-default_cli_parameters = DefaultCLIParameters()
+    password = "123456"
+    remote_path = "/sdcard/Android/data/com.mi.health/files/log/devicelog"
+    source_path = '~/Downloads'
```
Comment (Junbo-Zheng) [inline pyextract.py]:
```suggestion
    source_path = "~/Downloads"
```

File: pyextract.py [hunk 3]
```diff
@@ -45,55 +42,53 @@ def __init__(self):
...
             nargs=1,
-            help=
-            "extract packet filename, the default file suffix is .tar.gz, such as: log.tar.gz",
+            help="extract packet filename, the default file suffix is .tar.gz, such as: log.tar.gz",
             required=True)
         arg_parser.add_argument(
             '-p', '--purge_source_file',
-            help=
-            'purge source file if is true',
+            help='purge source file if is true',
```
Comment (Junbo-Zheng) [inline pyextract.py]:
```suggestion
            help="purge source file if is true",
```

File: pyextract.py [hunk 4]
```diff
@@ -45,55 +42,53 @@ def __init__(self):
...
             action='store_true',
             default=False)
         arg_parser.add_argument(
             '-F', '--filter_pattern',
             type=str,
-            default=default_cli_parameters.filter_pattern,
+            default=DefaultCLIParameters.filter_pattern,
             help='filter the files to be merged')
```
Comment (Junbo-Zheng) [inline pyextract.py]:
```suggestion
            help="filter the files to be merged")
```

File: pyextract.py [hunk 5]
```diff
@@ -102,139 +97,183 @@ def __set__(self, instance, value):
...
+    def __init__(self, cli_parser):
+        self.__cli_parser = cli_parser
+        self.log_packet_path = None
+        self.log_dir_path = None
+
+    def clear_output_dir(self, ask=True):
+        if not os.path.exists(self.__cli_parser.output_path):
+            return 0
+
+        # if exists output path, clear
```
Comment (Junbo-Zheng) [inline pyextract.py]:
```suggestion
        # if output path exists, clear
```

File: pyextract.py [hunk 6]
```diff
@@ -102,139 +97,183 @@ def __set__(self, instance, value):
...
-        if re.match(args.filter_pattern, file) is None:
-            continue
-        cmd += os.path.join(path, file) + " "
+class LogTools:
+    def __init__(self, cli_parser):
+        self.__cli_parser = cli_parser
+        self.log_packet_path = None
+        self.log_dir_path = None
+
+    def clear_output_dir(self, ask=True):
```
Comment (Junbo-Zheng) [inline pyextract.py:106]:
```suggestion
    def clear_output_dir(self, cover=True):
```

File: pyextract.py [hunk 7]
```diff
@@ -102,139 +97,183 @@ def __set__(self, instance, value):
...
+        if not os.path.exists(self.__cli_parser.output_path):
+            return 0
+
+        # if exists output path, clear
+        if ask:
```
Comment (Junbo-Zheng) [inline pyextract.py:111]:
```suggestion
        if cover:
```

File: pyextract.py [hunk 8]
```diff
@@ -102,139 +97,183 @@ def __set__(self, instance, value):
...
+            print(adb_cmd)
+            ShellRunner.command_run(adb_cmd)
+
+            file = os.getcwd() + "/devicelog/**/" + "*" + self.__cli_parser.filename[0] + "*.tar*.gz"
+            result = glob.glob(file, recursive=True)
+        else:
+            pattern = self.__cli_parser.source_path[0] + "/" + "*" + self.__cli_parser.filename[0] + "*.tar*.gz"
+            result = glob.glob(pattern)
+
+        print(Highlight.Convert("pull") + " %s from %s" % (result, self.__cli_parser.source_path[0]))
```
Comment (Junbo-Zheng) [inline pyextract.py]:
```
        print(Highlight.Convert("pull") + " %s from %s" % (result, self.__cli_parser.source_path[0]))
```
move to after line 138

File: pyextract.py [hunk 9]
```diff
@@ -102,139 +97,183 @@ def __set__(self, instance, value):
...
-        file = result[int(index)]
-    else:
-        file = result[0]
+        # remove unused .gz files
+        return self.__remove_all_suffix_gz_file__()
 
-    path, unused = os.path.split(file)
-    output = args.filename[0] + "_" + default_cli_parameters.output_file
-    output = os.path.join(path, output)
+    def merge_logfiles(self):
```
Comment (Junbo-Zheng) [inline pyextract.py:222]:
```suggestion
    # merge all files into a new file
    def merge_logfiles(self):
```

File: pyextract.py (hunk 6 above)
Comment (nowhszh) [reply pyextract.py:106]: Why is this modified

**Review by Junbo-Zheng — APPROVED** [review]

**Comment by Junbo-Zheng** [comment]
> @nowhszh Please update README.md too.

done #14

**Pull request merged**